newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -75,12 +78,44 @@ export default class AuthenticatedUserProvider extends React.Component<
         onEdit: () => this.openEditProfileDialog(true),
         onCreateAccount: () => this.openCreateAccountDialog(true),
         onRefreshUserProfile: this._fetchUserProfile,
+        onRefreshFirebaseProfile: this._reloadFirebaseProfile,
+        onSendEmailVerification: this._doSendEmailVerification,
         getAuthorizationHeader: () =>
           this.props.authentication.getAuthorizationHeader(),
       },
     });
   }
 
+  _reloadFirebaseProfile = () => {
+    const { authentication } = this.props;
+
+    authentication.getFirebaseUser((err, firebaseUser: ?FirebaseUser) => {
+      if (err && err.unauthenticated) {
+        return this.setState({
```

Comment:
Use `({authenticatedUser}) => ({authenticatedUser: {...authenticatedUser, ...}})` form.

This _reloadFirebaseProfile is exactly the beginning of `_fetchUserProfile`. I reckon we can implement `_fetchUserProfile` using it then (either by passing it an optional callback to be called with the `firebaseUser` or by reworking this to return a promise).